fhir-server/src/test/java/com/ibm/fhir/server/test/ServerResolveFunctionTest.java

```diff
@@ -0,0 +1,374 @@
+
+        SingleResourceResult<Observation> result = persistence.read(context, Observation.class, "54321");
+        assertNotNull(result);
+        assertNotNull(result.getResource());
+        assertTrue(result.isSuccess());
+
+        Observation observation = result.getResource();
+
+        FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();
+        EvaluationContext evaluationContext = new EvaluationContext(observation);
+
+        Collection<FHIRPathNode> nodes = evaluator.evaluate(evaluationContext, "subject.resolve() is Patient");
+        assertEquals(nodes, SINGLETON_TRUE);
+
+        nodes = evaluator.evaluate(evaluationContext, "subject.resolve().id = '12345'");
+        assertEquals(nodes, SINGLETON_TRUE);
+
+        nodes = evaluator.evaluate(evaluationContext, "subject.resolve().name.given = 'John'");
+        assertEquals(nodes, SINGLETON_TRUE);
+
+        nodes = evaluator.evaluate(evaluationContext, "subject.resolve().name.family = 'Doe'");
+        assertEquals(nodes, SINGLETON_TRUE);
+
+        nodes = evaluator.evaluate(evaluationContext, "subject.resolve().deceased = false");
+        assertEquals(nodes, SINGLETON_TRUE);
+
+        evaluationContext.setExternalConstant(ResolveFunction.RESOLVE_RELATIVE_REFERENCES, FHIRPathBooleanValue.FALSE);
+        nodes = evaluator.evaluate(evaluationContext, "subject.resolve() is Patient");
+        assertEquals(nodes, SINGLETON_TRUE);
+
+        nodes = evaluator.evaluate(evaluationContext, "subject.resolve().id = '12345'");
+        assertEquals(nodes, empty());
+
+        evaluationContext.setExternalConstant(ResolveFunction.RESOLVE_RELATIVE_REFERENCES, FHIRPathBooleanValue.TRUE);
+        nodes = evaluator.evaluate(evaluationContext, "subject.resolve().managingOrganization.resolve() is Organization");
+        assertEquals(nodes, SINGLETON_TRUE);
+
+        nodes = evaluator.evaluate(evaluationContext, "subject.resolve().managingOrganization.resolve().id = '67890'");
+        assertEquals(nodes, SINGLETON_TRUE);
+
+        nodes = evaluator.evaluate(evaluationContext, "subject.resolve().managingOrganization.resolve().name = 'Good Samaritan'");
+        assertEquals(nodes, SINGLETON_TRUE);
+    }
+
+    public static class PersistenceHelperImpl implements PersistenceHelper {
+        private final FHIRPersistence persistence;
+
+        public PersistenceHelperImpl() {
+            persistence = new PersistenceImpl();
+        }
+
+        @Override
+        public FHIRPersistence getFHIRPersistenceImplementation() throws FHIRPersistenceException {
+            return persistence;
+        }
+
+        @Override
+        public FHIRPersistence getFHIRPersistenceImplementation(String factoryPropertyName) throws FHIRPersistenceException {
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    public static class PersistenceImpl implements FHIRPersistence {
```

Comment:
This is like the 4th mock / stubbed out persistence impl we have across our various modules.  I wonder if its worth trying to consolidate.

Reply:
Yes, the one I made is map-based and could be generalized and perhaps reused elsewhere. That said, such consolidation is outside the scope of this PR.